src/main/java/com/theelixrlabs/healthcare/controller/DoctorController.java

```diff
@@ -59,4 +65,21 @@ public ResponseEntity<?> getDoctorById(@PathVariable(DoctorConstants.PATH_VARIAB
         DoctorDto doctorDto = doctorService.getDoctorById(doctorId);
         return new ResponseEntity<>(new SuccessResponse<>(true, doctorDto), HttpStatus.OK);
     }
+
+    /**
+     * Retrieves a list of doctors by their name.
+     *
+     * @param doctorName the name of the doctor to search for
+     * @return a ResponseEntity containing a SuccessResponse with a list of matching DoctorDto objects
+     */
+    @GetMapping(DoctorConstants.GET_DOCTORS_BY_NAME_ENDPOINT)
+    public ResponseEntity<SuccessResponse<List<DoctorDto>>> getDoctorsByName
+    (@RequestParam(DoctorConstants.DOCTOR_NAME_PARAM) String doctorName) {
+        doctorModelValidator.validateDoctorName(doctorName);
+        List<DoctorDto> doctorDtoList = doctorService.searchDoctorByName(doctorName);
+        if (doctorDtoList.isEmpty()) {
```

Comment:
All these empty or null checks move it in the service layer. controller method should accept requests and it should give a response structure

Reply:
yes changed it.